docs/contributing/commit-discipline.md

```diff
@@ -279,6 +279,63 @@ might worry about it breaking.
   you developed this commit or pull request, like "First I tried X" or
   "I changed Y".
 
+#### Don't @-mention people in commit messages
+
+**No GitHub @-mentions:** GitHub makes it tempting to refer to people
+by an @-mention of their GitHub username, like `@gnprice`. Unfortunately
+this is a misfeature in commit messages:
+
+- GitHub will turn this into a notification for the person every time
+  a version of the commit is rebased and pushed somewhere, which is
+  usually totally irrelevant for them -- e.g. someone took a long
+  stretch of history and rebased it and pushed that to their own
+  clone.
+
+  That person is doing something perfectly reasonable because it's off
+  in their own sandbox, and the author (you) had a perfectly
+  reasonable intention to mention the person when making a change in
+  the project, but GitHub turns the combination of those into spam.
+
+- It also ties the information unnecessarily to GitHub, which for a
+  long-lived project may not be its home forever.
+
+If you want to send someone a notification about a change, @-mention
+them in the PR thread, not in a commit message.
+
+To refer to them in a commit message, use one of the conventions
+below.
+
+**Consider just a name:** If there's further context -- like a chat
+link where you're referring to what someone said in that conversation
+-- then just a name like "Greg" can be natural and unambiguous.
+
+**Consider `Reported-by:` and friends:** The convention of the Linux
+kernel project, and the upstream Git project, is to end the commit
+message with a block of lines referring to people who wrote the
+commit, reviewed it, approved it, etc. These identify people by name
+and email address, for complete unambiguity.
+
+We don't have a habit of using this in our codebase, but that doesn't
+stop it from working just fine. For example, your commit message might
+have a line like:
+
+    Suggested-by: Anders Kaseorg <anders@zulipchat.com>
```

Comment:
Perhaps take the opportunity to update this, to `@zulip.com` rather than `@zulipchat.com`.